Fix OIDC tests.

vbabanin opened the pull request:
JAVA-5337

stIncMale commented on driver-sync/src/test/functional/com/mongodb/internal/connection/OidcAuthenticationProseTests.java:
```diff
@@ -268,14 +272,19 @@ private static Stream<Arguments> testValidCallbackInputsTimeoutWhenTimeoutMsIsSe
-            + "infinite and timeoutMs is set to {0}")
-    @ValueSource(ints = {0, 100})
-    void testCallbackTimeoutWhenServerSelectionTimeoutIsInfiniteTimeoutMsIsSet(final int timeoutMs) {
+    @Test
+    @DisplayName("test callback timeout when server selection timeout and "
+            + "timeoutMs are infinite")
+    void testCallbackTimeoutWhenServerSelectionTimeoutIsInfiniteTimeoutMsIsSet() {
```
`timeoutMs`/`server selection timeout` represent neither the names of the `ConnectionString` options, nor names of the API methods, which makes the naming both inconsistent with the existing test code, and undiscoverable.

```suggestion
    @Test
    @DisplayName("test callback timeout when serverSelectionTimeoutMS and timeoutMS are infinite")
    void testCallbackTimeoutWhenServerSelectionTimeoutMsIsInfiniteTimeoutMsIsSet() {
```

stIncMale commented on driver-sync/src/test/functional/com/mongodb/internal/connection/OidcAuthenticationProseTests.java:
```diff
@@ -268,14 +272,19 @@ private static Stream<Arguments> testValidCallbackInputsTimeoutWhenTimeoutMsIsSe
 
         OidcCallback callback2 = (context) -> {
-            assertEquals(context.getTimeout(), ChronoUnit.FOREVER.getDuration());
+            assertEquals(expectedTimeout, context.getTimeout(),
+                    format("Expected timeout to be infinite (%d seconds), but was %d seconds",
+                            expectedTimeout.getSeconds(),
+                            context.getTimeout().getSeconds()));
```
Why are we not just relying on `Duration.toString` here?

```suggestion
            assertEquals(expectedTimeout, context.getTimeout(),
                    format("Expected timeout to be infinite (%s), but was %s",
                            expectedTimeout, context.getTimeout()));
```

stIncMale reviewed (changes requested):
It took some time to figure out why the only PR for https://jira.mongodb.org/browse/JAVA-5337 is the current one, where just a few test code lines are changed. Turned out that the commit containing the actual changes https://github.com/mongodb/mongo-java-driver/commit/68d5421f1af2d932509e1df862adef6440bad927 was merged with an incorrect description linking to https://jira.mongodb.org/browse/JAVA-5357. This will make the ticket undiscoverable in the future.

vbabanin commented:
> It took some time to figure out why the only PR for https://jira.mongodb.org/browse/JAVA-5337 is the current one, where just a few test code lines are changed. Turned out that the commit containing the actual changes [68d5421](https://github.com/mongodb/mongo-java-driver/commit/68d5421f1af2d932509e1df862adef6440bad927) was merged with an incorrect description linking to https://jira.mongodb.org/browse/JAVA-5357. This will make the ticket undiscoverable in the future.

I’ve added cross-references in both tickets and updated the PR comments for traceability. I’ll also check what can be done to unlink the incorrect PR association from the ticket.
